src/main/scala/scalaz/reactive/Event.scala

```diff
@@ -1,11 +1,37 @@
 package scalaz.reactive
+import scalaz.Monoid
+import scalaz.zio.{ IO, RTS }
 
-case class Event[+A](value: Future[Reactive[A]]) extends AnyVal {
+case class Event[A](value: Future[Reactive[A]]) extends RTS { self =>
+
+  def merge(
+    v: Event[A]
+  ): Event[A] = { // if Event[+A]: covariant type A occurs in contravariant position
+
+    val winner: IO[Nothing, Either[(Time, Reactive[A]), (Time, Reactive[A])]] =
+      self.value.force.map(Left(_)).race(v.value.force.map(Right(_)))
+
+    val io: IO[Nothing, Future[Reactive[A]]] = winner.map(_ match {
+      case Left((t, r)) =>
+        Future((t, Reactive(r.head, r.tail)))
+      case Right((t, r)) =>
+        Future((t, Reactive(r.head, r.tail)))
+    })
+
+    Event(unsafeRun(io)) // That is not good
```

Comment:
Indeed it is not good, we should NEVER use the `RTS` trait on our code, or we will be limiting our implementation to the jvm and it is clear that we are leaking side effects. this is why I said we can experiment with `type Future[+A] = IO[Void, (Time, A)]` so that in this case it is a simple race operation:
```
def merge[AA >: A](v: Event[AA]): Event[AA] = Event(self.value race v.value)
```

Reply:
wow :-)

Reply:
Indeed, once we are in IO, we have to stay in the IO, so I can't think of another way to do this at the moment.